File: app/src/main/java/com/iavariav/kbmonline/ui/atasan/presenter/HistoriAtasanPresenter.java
package com.iavariav.kbmonline.ui.atasan.presenter;

import android.content.Context;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.Toast;

import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.iavariav.kbmonline.model.PemesananModel;
import com.iavariav.kbmonline.rest.ApiConfig;
import com.iavariav.kbmonline.rest.ApiService;
import com.iavariav.kbmonline.ui.atasan.adapter.AtasanHistoriAprovalAdapter;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class HistoriAtasanPresenter {
    private ArrayList<PemesananModel> pemesananModels;
    private AtasanHistoriAprovalAdapter atasanHistoriAprovalAdapter;

    public void getDatas(final Context context, String idUser, final RecyclerView recyclerView, final LinearLayout div){
        pemesananModels = new ArrayList<>();
        ApiService apiService = ApiConfig.getApiService();
        apiService.getDataHistoriAtasan("getHistoriAtasan", idUser)
                .enqueue(new Callback<ArrayList<PemesananModel>>() {
                    @Override
                    public void onResponse(Call<ArrayList<PemesananModel>> call, Response<ArrayList<PemesananModel>> response) {
                        if (response.isSuccessful()){
                            pemesananModels = response.body();
                            atasanHistoriAprovalAdapter = new AtasanHistoriAprovalAdapter(context, pemesananModels);
                            recyclerView.setLayoutManager(new LinearLayoutManager(context));
                            recyclerView.setAdapter(atasanHistoriAprovalAdapter);
                            atasanHistoriAprovalAdapter.notifyDataSetChanged();


                            if (pemesananModels.isEmpty()) {
                                recyclerView.setVisibility(View.GONE);
                                div.setVisibility(View.VISIBLE);
                            } else {
                                recyclerView.setVisibility(View.VISIBLE);
                                div.setVisibility(View.GONE);
                            }
                        }
                    }

                    @Override
                    public void onFailure(Call<ArrayList<PemesananModel>> call, Throwable t) {
                        Toast.makeText(context, "" + t.getMessage(), Toast.LENGTH_SHORT).show();

                    }
                });

    }
}

File: app/src/main/java/com/iavariav/kbmonline/ui/atasan/presenter/HistoriAtasanPresenter.kt
package com.iavariav.kbmonline.ui.atasan.presenter

import android.view.View
import android.widget.LinearLayout
import android.widget.Toast
import androidx.fragment.app.FragmentActivity

import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

import com.iavariav.kbmonline.model.PemesananModel
import com.iavariav.kbmonline.rest.ApiConfig
import com.iavariav.kbmonline.ui.atasan.adapter.AtasanHistoriAprovalAdapter

import java.util.ArrayList

import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class HistoriAtasanPresenter {
    private var pemesananModels: ArrayList<PemesananModel>? = null
    private var atasanHistoriAprovalAdapter: AtasanHistoriAprovalAdapter? = null

    fun getDatas(context: FragmentActivity?, idUser: String?, recyclerView: RecyclerView?, div: LinearLayout?) {
        pemesananModels = ArrayList()
        val apiService = ApiConfig.apiService
        apiService.getDataHistoriAtasan("getHistoriAtasan", idUser)
                .enqueue(object : Callback<ArrayList<PemesananModel>> {
                    override fun onResponse(call: Call<ArrayList<PemesananModel>>, response: Response<ArrayList<PemesananModel>>) {
                        if (response.isSuccessful) {
                            pemesananModels = response.body()
                            atasanHistoriAprovalAdapter = AtasanHistoriAprovalAdapter(context, pemesananModels!!)
                            recyclerView?.layoutManager = LinearLayoutManager(context)
                            recyclerView?.adapter = atasanHistoriAprovalAdapter
                            atasanHistoriAprovalAdapter!!.notifyDataSetChanged()


                            if (pemesananModels!!.isEmpty()) {
                                recyclerView?.visibility = View.GONE
                                div?.visibility = View.VISIBLE
                            } else {
                                recyclerView?.visibility = View.VISIBLE
                                div?.visibility = View.GONE
                            }
                        }
                    }

                    override fun onFailure(call: Call<ArrayList<PemesananModel>>, t: Throwable) {
                        Toast.makeText(context, "" + t.message, Toast.LENGTH_SHORT).show()

                    }
                })

    }
}
